content/blog/dbplyr-2-0-0/index.Rmd

```diff
@@ -0,0 +1,204 @@
+author: Hadley Wickham
+description: >
+    dbplyr 2.0.0 brings dplyr 1.0.0 compatibility, numeric improvements
+    to SQL translation (including new Amazon Redshift and SAP HANA 
+    backends), and an improved system for extending dbplyr to work with
+    other databases.
+    
+photo:
+  url: https://unsplash.com/photos/r2A6WYI8YIg
+  author: Shawn Ang
+
+categories: [package] 
+tags: [dbplyr, dplyr]
+---
+
+```{r, include = FALSE}
+library(dplyr, warn.conflicts = FALSE)
+```
+
+We're pleased to announce the release of [dbplyr](https://dbplyr.tidyverse.org/) 2.0.0.
+dbplyr is a database backend for [dplyr](https://dplyr.tidyverse.org/) that allows you to use a remote database as if it was a collection of local data frames: you write ordinary dplyr code and dbplyr translates it to SQL for you.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("dbplyr")
+```
+
+This blog post covers the major improvements in this version:
+
+-   dplyr 1.0.0 compatibility so you can now use `across()`, `relocate()`, `rename_with()`, and more.
+
+-   The major improvements to SQL translation.
+
+-   A snazzy new logo from [Allison Horst](https://www.allisonhorst.com).
+
+-   An improved extension system.
+
+Please see the [release notes](https://github.com/tidyverse/dbplyr/releases/tag/v2.0.0) for a full list of changes.
+
+```{r setup, message = FALSE}
+library(dbplyr)
+library(dplyr, warn.conflicts = FALSE)
+```
+
+## dplyr 1.0.0 compatibility
+
+dbplyr now supports all relevant features added in dplyr 1.0.0:
+
+-   `across()` is now translated into individual SQL statements.
+
+    ```{r}
+    lf <- lazy_frame(g = 1, a = 1, b = 2, c = 3)
+    lf %>% 
+      group_by(g) %>% 
+      summarise(across(everything(), mean, na.rm = TRUE))
+    ```
+
+-   `rename()` and `select()` support dplyr 1.0.0 tidyselect syntax, apart from predicate functions which can't easily work on computed queries.
+    You can now use `rename_with()` to programmatically rename columns.
+
+    ```{r}
+    lf <- lazy_frame(x1 = 1, x2 = 2, x3 = 3, y1 = 4, y2 = 3)
+    lf %>% select(starts_with("x") & !"x3")
+    lf %>% select(ends_with("2") | ends_with("3"))
+    lf %>% rename_with(toupper)
+    ```
+
+-   `relocate()` makes it easy to move columns around:
+
+    ```{r}
+    lf <- lazy_frame(x1 = 1, x2 = 2, y1 = 4, y2 = 3)
+    lf %>% relocate(starts_with("y"))
+    ```
+
+-   `slice_min()`, `slice_max()`, and `slice_sample()` are now supported, and `slice_head()` and `slice_tail()` throw informative error messages (since they don't make sense for databases).
+
+    ```{r}
+    lf <- lazy_frame(g = rep(1:2, 5), x = 1:10)
+    lf %>% 
+      group_by(g) %>% 
+      slice_min(x, prop = 0.5)
+
+    lf %>% 
+      group_by(g) %>% 
+      slice_sample(x, n = 10, with_ties = TRUE)
+    ```
+
+    Note that these slices are translated to window functions, and because you can't use a window function directly inside a `WHERE` clause, they must be wrapped in a subquery.
```

Comment:
```suggestion
    Note that these slices are translated into window functions, and because you can't use a window function directly inside a `WHERE` clause, they must be wrapped in a subquery.
```

Reply:
Translated _into_ is (evidently) 200x more common in English than translated _to_. I gave you one "translated to", but I'm changing the next 199.
<https://jakubmarian.com/translate-to-vs-translate-into-in-english/>